pallets/loans/src/lib.rs

```diff
@@ -1011,6 +1209,188 @@ impl<T: Config> Pallet<T> {
         Ok(total_asset_value)
     }
 
+    fn reward_scale() -> u128 {
+        10_u128.pow(12)
+    }
+
+    fn calculate_reward_delta_index(
+        delta_block: T::BlockNumber,
+        reward_speed: BalanceOf<T>,
+        total_share: BalanceOf<T>,
+    ) -> Result<u128, sp_runtime::DispatchError> {
+        let delta_block: BalanceOf<T> = delta_block.saturated_into();
+        let delta_index = delta_block
+            .get_big_uint()
+            .checked_mul(&reward_speed.get_big_uint())
+            .and_then(|r| r.checked_mul(&Self::reward_scale().get_big_uint()))
+            .and_then(|r| r.checked_div(&total_share.get_big_uint()))
+            .and_then(|r| r.to_u128())
+            .ok_or(ArithmeticError::Overflow)?;
+        Ok(delta_index)
+    }
+
+    fn calculate_reward_delta(
+        share: BalanceOf<T>,
+        reward_delta_index: u128,
+    ) -> Result<u128, sp_runtime::DispatchError> {
+        let reward_delta = share
+            .get_big_uint()
+            .checked_mul(&reward_delta_index.get_big_uint())
+            .and_then(|r| r.checked_div(&Self::reward_scale().get_big_uint()))
+            .and_then(|r| r.to_u128())
+            .ok_or(ArithmeticError::Overflow)?;
+        Ok(reward_delta)
+    }
+
+    fn update_reward_supply_index(asset_id: AssetIdOf<T>) -> DispatchResult {
+        let current_block_number = <frame_system::Pallet<T>>::block_number();
+        RewardSupplyState::<T>::try_mutate(asset_id, |supply_state| -> DispatchResult {
+            let delta_block = current_block_number.saturating_sub(supply_state.block);
+            if !delta_block.is_zero() {
+                let supply_speed = MarketRewardSpeed::<T>::get(asset_id);
+                if supply_speed > 0 {
```

Comment:
should be consistent, maybe `!..is_zero` is better?

Reply:
ok